iml-agent/src/action_plugins/check_stonith.rs

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    agent_error::{CibError, ImlAgentError},
+    cmd::cmd_output,
+};
+use elementtree::Element;
+use futures::Future;
+use iml_wire_types::{ComponentState, ConfigState, ServiceState, RunState};
+use std::default::Default;
+use std::process::Command;
+use tracing::{span, Level};
+use tracing_futures::Instrument;
+
+fn inst_attr_problem(elem: &Option<&Element>, node: &String) -> Option<String> {
+    match elem {
+        None => None,
+        Some(instance) => {
+            let mut hc: Option<bool> = None;
+            let mut hl: Option<bool> = None;
+            for e in instance.children() {
+                match e.get_attr("name") {
+                    Some("pcmk_host_list") => match e.get_attr("value") {
+                        None => return None,
+                        Some(val) => {
+                            if val.contains(node) {
+                                match hc {
```

Comment:
I might be reading this incorrectly but it looks like  `hc` will always be `None` right?

Reply:
No, It's a big set of match statements, hc and hl can be set in either order, I'll add a comment to make it clearer